application/client.core/src/app/environment/components/sidebar/shell/running/component.ts

```diff
@@ -72,21 +78,13 @@ export class SidebarAppShellRunningComponent implements OnDestroy, OnInit {
         };
     }
 
-    public _ng_onContexMenu(event: MouseEvent, process: IPCMessages.IShellProcess) {
-        const items: IMenuItem[] = [
-            {
-                caption: 'Terminate',
-                handler: () => {
-                    this.service.terminate({ session: this._sessionID, guid: process.guid }, process.command ).catch((error: string) => {
-                        this._logger.error(error);
-                    });
-                },
-            },
-        ];
-        ContextMenuService.show({
-            items: items,
-            x: event.pageX,
-            y: event.pageY,
+    public _ng_onTerminate(process: IPCMessages.IShellProcess) {
+        this.service.terminate({ session: this._sessionID, guid: process.guid }, process.command ).catch((error: Error) => {
+            this._showNotification({
+                caption: 'Failed to terminate process',
+                message: error.message,
+            });
+            this._logger.error(error);
```

Comment:
I would ask you do not pass into logs whole error object. Instead this
```
this._logger.error(error);
```
use please this
```
this._logger.error(error.message);
```
Because `errors` is an object, logger will try to convert it into string and we will get huge output with all properties.

Also, take in account, you can refactor a little your code to this:
```
this._showNotification({
    caption: 'Failed to terminate process',
    message: this._logger.error(error.message),
});
```
Well `logger` always returns same string as you passed into. It allows you make code more compact.